lib/ansible/modules/cloud/azure/azure_rm_deployment.py

```diff
@@ -56,6 +56,7 @@
     description:
       - A hash containing the templates inline. This parameter is mutually exclusive with 'template_link'.
         Either one of them is required if "state" parameter is "present".
+    type: object
```

Comment:
`object` is not a valid type. This is meant to reflect the type specified in the module argument spec, which is currently `dict`. I'm not sure setting the type to `dict` will fix the external validation issue you're seeing, but this would make the module documentation inaccurate.

Reply:
My mistake. I changed it to dict and posted an issue here: https://github.com/shaded-enmity/ansible-schema-generator/issues/1 asking if the schema generator can make the mapping to object.

Reply:
Now I must admit _my_ mistake. This field is not really at all what I thought it was. After getting it explained to me by some other folks (and reading the docs), the only valid value in this field is `type: bool`. From the [docs](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html):

> type: If an argument is type='bool', this field should be set to type: bool and no choices should be specified.

This is only used to improve the way docs are built. It does not validate the type in the `argument_spec` against the type here as I (incorrectly) imagined in my head. 😊